src/pages/api/create-job.ts

```diff
@@ -0,0 +1,279 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import * as z from "zod";
+import { ImmutableHeaders, ImmutableURL } from "immurl";
+import dedent from "dedent";
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method !== "POST") {
+    return res.status(405).end();
+  }
+  const bodyParsedResult = bodySchema.safeParse(req.body);
+
+  if (!bodyParsedResult.success) {
+    return res.status(400).json(bodyParsedResult.error);
+  }
+
+  const gh = new GithubAPI.Client(
+    constants.GITHUB_AUTH_USERNAME,
+    constants.GITHUB_AUTH_TOKEN
+  );
+
+  const branchName = `jobs/${new Date().toISOString().replace(/[:\.]/g, "_")}`;
+
+  const repo = {
+    owner: constants.GITHUB_OWNER,
+    repo: constants.GITHUB_REPO,
+  } as const;
+
+  const masterRef = await gh.getRef({
+    ...repo,
+    ref: `heads/${constants.GITHUB_BASE_BRANCH}`,
+  });
+
+  // For some reason this produces a 404? WTF
+  const prBranchRef = await gh.createRef({
+    ...repo,
+    ref: `refs/heads/${branchName}`,
+    sha: masterRef.object.sha,
+  });
+
+  const createFileResult = await gh.createFile({
+    ...repo,
+    content: Buffer.from(getJobMarkdownContent(bodyParsedResult.data)).toString(
+      "base64"
+    ),
+    message: "Add job",
+    path: `src/pages/${new Date().toISOString()}/index.mdx`,
```

Comment:
Just had a thought:
I think it would be safer to use regular `.md` files for this.

Not validated it but it's possible to embed arbitrary JS in MDX files. This combined with auto deploys for PRs could be a way to do some nasty stuff via the next preview deploys...

I'm assuming that this is all mitigated by using regular markdown.